scalafmt-core/shared/src/main/scala/org/scalafmt/util/LiteralOps.scala

```diff
@@ -0,0 +1,70 @@
+package org.scalafmt.util
+
+import org.scalafmt.config.{Case, ScalafmtConfig}
+
+object LiteralOps {
+
+  /**
+    * Prints integer literals with specified case
+    *
+    * Divides literal into three parts:
+    *
+    * 0xFF123L
+    * 0x is a hex prefix
+    * FF123 is a body
+    * L is a long suffix
+    *
+    * literals.long applies to suffix and literals.hex applies to body
+    * */
+  def prettyPrintInteger(
+      str: String
+  )(implicit style: ScalafmtConfig): String = {
+    val sb = new StringBuilder()
```

Comment:
why not pass the string builder from format writer and append to that directly?

and while we're at it, change the Case.process method to do the same?

Reply:
I commited https://github.com/scalameta/scalafmt/pull/1857/commits/d0bc2a1397698042efea9998892699dc6b366516 with respecting `literal.hex` for `0x` prefix, and this function greatly simplified. So there is no more need for `StringBuilder`